src/Pos/Block/Network/Retrieval.hs

```diff
@@ -57,68 +57,103 @@ retrievalWorker getPeers = worker outs (retrievalWorkerImpl getPeers)
                              (Proxy :: Proxy (MsgBlock ssc))
                       ]
 
+-- I really don't like join
+{-# ANN retrievalWorkerImpl ("HLint: ignore Use join" :: Text) #-}
+
+-- | Worker that queries blocks. It has two jobs:
+--
+-- * If there are headers in 'ncBlockRetrievalQueue', this worker retrieves
+--   blocks according to that queue.
+--
+-- * If recovery is in progress, this worker keeps recovery going.
```

Comment:
... by asking headers and backing up to (1 ~ block retrieval) on the next loop iteration. Please add.